pkg/webservice/handlers.go

```diff
@@ -339,3 +342,55 @@ func getClusterConfig(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+var mutex sync.Mutex
+func updateConfig(w http.ResponseWriter, r *http.Request) {
+	mutex.Lock()
+	defer mutex.Unlock()
+	writeHeaders(w)
+	requestBytes, err := ioutil.ReadAll(r.Body)
+
+	if err == nil {
```

Comment:
when `err != nil` we should log an error in the server-side and response the client with an error too.,

Reply:
I added a log message when creating the response. Also handled this case as well.